xtraplatform-crs/src/main/java/de/ii/xtraplatform/crs/infra/CrsTransformerFactoryProj.java

```diff
@@ -85,8 +86,16 @@ public boolean isCrsSupported(EpsgCrs crs) {
             crsCache.computeIfAbsent(crs, mayThrow(ignore -> {
                 String code = String.valueOf(applyWorkarounds(crs).getCode());
                 CoordinateReferenceSystem coordinateReferenceSystem = crsFactory
-                    .createCoordinateReferenceSystem(code);
-                return applyAxisOrder(coordinateReferenceSystem, crs.getForceAxisOrder());
+                        .createCoordinateReferenceSystem(code);
+                coordinateReferenceSystem = applyAxisOrder(coordinateReferenceSystem, crs.getForceAxisOrder());
+                if (crs.getVerticalCode().isPresent()) {
+                    String verticalCode = String.valueOf(crs.getVerticalCode().getAsInt());
+                    CoordinateReferenceSystem verticalCrs = crsFactory.createVerticalCRS(verticalCode);
+                    return new DefaultCompoundCRS("", coordinateReferenceSystem, verticalCrs);

```

Comment:
From my understanding, a DefaultCompoundCRS object is not created because of an exception in javax.measure library, which is not directly related to CRS values used in the test. The exception is thrown in javax.measure.unit.Unit::asType:172 after being called at javax.measure.unit.NonSI:667.

Reply:
Sorry, DefaultCompoundCRS is from geotools, that should not be used. I pushed a fix, the CompoundCRS is now created correctly.